packages/react/ds/src/progress-stepper/progress-stepper.tsx

```diff
@@ -0,0 +1,78 @@
+import { Icon } from '../icon/icon.js';
+
+type Orientation = 'vertical' | 'horizontal';
+
+export type ProgressStepperProps = {
+  steps: string[];
+  currentStepIndex: number;
+  orientation?: Orientation;
+};
+
+export type StepProps = {
+  children: string;
+  isCurrentStep: boolean;
+  isCompleted: boolean;
+  stepNumber: number;
+  orientation?: Orientation;
+};
+
+const Step = ({
+  children,
+  isCurrentStep,
+  isCompleted,
+  stepNumber,
+  orientation,
+}: StepProps) => {
+  const isNextStep = !isCompleted && !isCurrentStep;
+  return (
+    <div className="gi-flex gi-relative gi-flex-1 gi-min-h-20 gi-min-w-20">
+      <div
+        className="gi-progress-stepper-step-container"
+        data-orientation={orientation}
+        data-current={isCurrentStep}
+        data-completed={isCompleted}
+        data-next={isNextStep}
+      >
+        <div className="gi-progress-stepper-step">
+          {isCompleted ? <Icon icon="check" /> : <div>{stepNumber}</div>}
+        </div>
+        {children}
```

Comment:
Should we limit the number of chars?
Maybe let the content break on two lines if it exceeds an X number of chars and then ellipsis.

For example:
If you have a longer string like "This step is for requesting information."
we can render something like
```
This step is for
requesting info...
```

At least we don't risk to break the UI if the user put a longer text.

Reply:
good call, it makes sense, working on this.